test/e2e/nodepool_upgrade_test.go

```diff
@@ -53,6 +53,10 @@ func NewNodePoolInPlaceUpgradeTestManifest(hostedCluster *hyperv1.HostedCluster,
 }
 
 func (ipu *NodePoolInPlaceUpgradeTestManifest) BuildNodePoolManifest(defaultNodepool hyperv1.NodePool) (*hyperv1.NodePool, error) {
+	isCompatible, err := e2eutil.ValidateNodePoolVersionCompatibility(ipu.previousReleaseImage, ipu.hostedCluster.Spec.Release.Image)
```

Comment:
why do we need this check here? we own this tests, we want to deterministically make them compatible or error otherwise